libipl/p10/ipl0.C

```diff
@@ -374,22 +385,106 @@ static int ipl_alignment_check(void)
 	return -1;
 }
 
+static int initialize_and_check_clock_chip(uint8_t& clock_count)
+{
+	struct pdbg_target *clock_target;
+	enum pdbg_target_status status;
+	uint8_t data;
+	int rc = 0;
+
+	// initialize output variable.
+	clock_count = 0;
+	
+	ipl_log(IPL_DEBUG, "Istep: soft reset clock, and verify clock status register\n");
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		status = pdbg_target_probe(clock_target);
+		assert(status == PDBG_TARGET_ENABLED);
+
+		// Resetting the clock chip, so that it will recalibrate input oscilator
+		// signal and identifies bad oscilators.
+		data = OSC_RESET_CMD;
+		rc = i2c_write(clock_target, 0, OSC_CTL_OFFSET, OSC_RESET_CMD_LEN, &data);
+		if(rc){
+			ipl_log(IPL_ERROR, "soft reset command is failed for clock %d, with rc = %d\n",
+				pdbg_target_index(clock_target), rc);
+
+			ipl_error_callback(IPL_ERR_CLK_FAILED);
```

Comment:
for callout purpose any position information is required? . what is the plab for adding callout here?

Reply:
i will push another commit on top of this to support callout